src/m365/spo/commands/file/file-sharinginfo-get.ts

```diff
@@ -0,0 +1,171 @@
+import * as chalk from 'chalk';
+import request from '../../../../request';
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import Utils from '../../../../Utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+import {FileSharingPrincipalType } from './FileSharingPrincipalType';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  id?: string;
+  url?: string;
+}
+
+class SpoFileSharinginfoGetCommand extends SpoCommand {
+
+  public get name(): string {
+    return commands.FILE_SHARINGINFO_GET;
+  }
+
+  public get description(): string {
+    return 'Generates the sharing information report for specified file';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = (!(!args.options.id)).toString();
+    telemetryProps.url = (!(!args.options.url)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.log(`Retrieving sharing information report for the file...`);
+    }
+
+    this.getneededFileInformation(args, logger)
+      .then((fileInformation: { fileItemId: number; libraryName: string; }): Promise<string> => {
+        if (this.verbose) {
+          logger.log(`Retrieving Sharing information report for the file with item Id  ${fileInformation.fileItemId}`);
+        }
+
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_api/web/lists/getbytitle('${fileInformation.libraryName}')/items(${fileInformation.fileItemId})/GetSharingInformation?$select=permissionsInformation&$Expand=permissionsInformation`,
+          headers: {
+            'accept': 'application/json;odata=nometadata'
+          },
+          responseType: 'json'
+        };
+        return request.post(requestOptions)
+      }).then((res: any): void => {
```

Comment:
Thanks for clarifying that point. I had the same doubt whether we need to have an interface created and infact I had created one before the first draft. But since I did not see the re usability of the same, I fell back to the `any` object. One difference was, I had that defined as a separate file which was not needed in first place. Now things are clear now.

What you have done seems to be a cleaner code and much better one. Looking awesome 👍

Reply:
Typically the way we go about it is, that if we pass the response as-is to output, we don't need to define a shape of the response, since we're not referring to any properties in our code. As we do however, having a type will help us ensure that we won't have any typos or anything else that would break on runtime.